builds/jenkins-build-pack-sls-app/Jenkinsfile

```diff
@@ -0,0 +1,942 @@
+
+      } //end of withCredentials
+    } //end of deployment to an environment
+  }
+}
+
+def deleteAndRedeployService(environment_logical_id, envBucketKey, accountDetails, config){
+  def s3bucketvalue = getS3BucketValue(accountDetails , config, envBucketKey)
+  echoServerlessFile()
+  sh "serverless remove --stage $environment_logical_id -v --bucket ${s3bucketvalue}"
+  def redeployOutput = servelessDeploy(environment_logical_id, envBucketKey)
+  if (redeployOutput != 'success') {
+    echo "Exception occured while serverless deployment to ${environment_logical_id} environment : $redeployOutput"
+    error "Exception occured while serverless deployment to ${environment_logical_id} environment"
+  }
+}
+
+def echoServerlessFile() {
+  def serverlessyml = readFile('serverless.yml').trim()
+  echo "serverless file data $serverlessyml"
+}
+
+def servelessDeploy(env, envBucketKey){
+  try {
+    def s3bucketvalue = getS3BucketValue(accountDetails, config, envBucketKey)
+    sh "serverless deploy --stage ${env} -v --bucket ${s3bucketvalue} > output.log"
+    def outputLog = readFile('output.log').trim()
+    echo "serverless deployment log $outputLog"
+    return outputLog
+  } catch (ex) {
+    echo "Serverless deployment failed due to $ex"
+    def outputLog = readFile('output.log').trim()
+    echo "serverless deployment log $outputLog"
+    throw ex
+  }
+}
+
+def removeEventResources(){
+  sh "sed -i -- '/#Start:resources/,/#End:resources/d' ./serverless.yml"
+  sh "sed -i -- '/#Start:events/,/#End:events/d' ./serverless.yml"
+}
+
+/**
+** Get BucketName by region
+**/
+def getS3BucketValue(accountDetails , config, env_bucketKey){
+  def s3bucketvalue
+  for (item in accountDetails.REGIONS) {
+    if(item.REGION == config.region){
+      s3bucketvalue = item.S3[env_bucketKey]
+    }
+  }
+  return s3bucketvalue
+}
+
+
+
+
+/**
+ */
+def isEnabled(config, key) {
+  if (config.containsKey(key)) {
+    return true
+  } else {
+    return false
+  }
+}
+
+def LoadConfiguration() {
+  def result = readFile('deployment-env.yml').trim()
+  echo "result of yaml parsing....$result"
+  def prop = [:]
+  def resultList = result.tokenize("\n")
+
+  // delete commented lines
+  def cleanedList = []
+  for (i in resultList) {
+    if (i.toLowerCase().startsWith("#")) { } else {
+      cleanedList.add(i)
+    }
+  }
+  // echo "result of yaml parsing after clean up....$cleanedList"
+  for (item in cleanedList) {
+    // Clean up to avoid issues with more ":" in the values
+    item = item.replaceAll(" ", "").replaceFirst(":", "#");
+    def eachItemList = item.tokenize("#")
+    //handle empty values
+    def value = null;
+    if (eachItemList[1]) {
+      value = eachItemList[1].trim();
+    }
+
+    if (eachItemList[0]) {
+      prop.put(eachItemList[0].trim(), value)
+    }
+
+  }
+  echo "Loaded configurations....$prop"
+  return prop
+}
+
+/**	Build project based on runtime
+ */
+def buildSlsApp(String runtime, String repo_name) {
```

Comment:
The class code itself don't add to the size much. It is usually the size of the library jars to support the function that add to the size.. unless the underlying jars used by each of the functions are disparate, it shouldn't make much of a difference. Valid point though but I think it can be ignored for now.